src/khoj/processor/plaintext/plaintext_to_jsonl.py

```diff
@@ -0,0 +1,116 @@
+# Standard Packages
+import glob
+import logging
+from pathlib import Path
+from typing import List
+
+# Internal Packages
+from khoj.processor.text_to_jsonl import TextToJsonl
+from khoj.utils.helpers import get_absolute_path, timer
+from khoj.utils.jsonl import load_jsonl, compress_jsonl_data
+from khoj.utils.rawconfig import Entry
+
+
+logger = logging.getLogger(__name__)
+
+
+class PlaintextToJsonl(TextToJsonl):
+    # Define Functions
+    def process(self, previous_entries=[]):
+        # Extract required fields from config
+        input_jsonl_files, input_jsonl_filter, output_file = (
+            self.config.input_files,
+            self.config.input_filter,
+            self.config.compressed_jsonl,
+        )
+
+        # Get Plaintext Input Files to Process
+        all_input_plaintext_files = PlaintextToJsonl.get_plaintext_files(input_jsonl_files, input_jsonl_filter)
+
+        # Extract Entries from specified plaintext files
+        with timer("Parse entries from plaintext files", logger):
+            current_entries = PlaintextToJsonl.convert_plaintext_entries_to_maps(
+                PlaintextToJsonl.extract_plaintext_entries(all_input_plaintext_files)
+            )
+
+        # Split entries by max tokens supported by model
+        with timer("Split entries by max token size supported by model", logger):
+            current_entries = self.split_entries_by_max_tokens(current_entries, max_tokens=256)
+
+        # Identify, mark and merge any new entries with previous entries
+        with timer("Identify new or updated entries", logger):
+            entries_with_ids = TextToJsonl.mark_entries_for_update(
+                current_entries, previous_entries, key="compiled", logger=logger
+            )
+
+        with timer("Write entries to JSONL file", logger):
+            # Process Each Entry from All Notes Files
+            entries = list(map(lambda entry: entry[1], entries_with_ids))
+            plaintext_data = PlaintextToJsonl.convert_entries_to_jsonl(entries)
+
+            # Compress JSONL formatted Data
+            compress_jsonl_data(plaintext_data, output_file)
+
+        return entries_with_ids
+
+    @staticmethod
+    def get_plaintext_files(jsonl_files=None, jsonl_file_filters=None):
+        "Get all jsonl files to process"
```

Comment:
These should say plaintext rather than jsonl I believe
```suggestion
    def get_plaintext_files(plaintext_files=None, plaintext_file_filters=None):
        "Get all plaintext files to process"
```